### PR TITLE
feat(atom/rss/json): support language and on feed and feed items (#196)

### LINKED ISSUE #196: language should be supported in atom feeds
According to the spec (https://validator.w3.org/feed/docs/atom.html), language can be supported using `xml:lang`. Unfortunately, this project here does only support it for RSS 2.0 feeds apparently.

### PR DESCRIPTION
**Pull request by marcojakob**
### Overview

This PR implements comprehensive language support across all feed formats (Atom 1.0, JSON Feed 1.1, and RSS 2.0), addressing issue #196 which requested language support in Atom feeds. The implementation goes beyond the original request to provide consistent language functionality across all supported feed formats.

#### Atom 1.0 Language Support

- **Feed-level language**: Added `xml:lang` attribute to `<feed>` element when language is specified
- **Per-item language**: Individual `<entry>` elements can have their own `xml:lang` attributes
- **Security**: Language values are properly sanitized to prevent XML injection attacks
- **Standards compliance**: Follows Atom 1.0 specification for language declarations

#### JSON Feed 1.1 Enhancement

- **Upgraded to JSON Feed 1.1**: Updated from version 1.0 to support the `language` field
- **Feed-level language**: Added `language` field to top-level feed object
- **Per-item language**: Individual items can specify their own language
- **Backward compatibility**: Maintains compatibility while adding new features

#### RSS 2.0 Verification
- **Confirmed existing support**: RSS 2.0 already had channel-level `<language>` element support
- **Added test coverage**: Comprehensive tests for RSS language functionality
- **Documented limitations**: RSS 2.0 spec doesn't support per-item language (by design)

#### Type System Updates

- Added `language?: string` to the `Item` interface for per-item language support
- Existing `FeedOptions.language` field was already present and working

#### Security Enhancements

- Language attributes are sanitized using the existing `sanitize()` function
- Prevents XML/XSS injection attacks via malformed language codes

#### Standards Compliance

- **Atom**: Uses `xml:lang` attributes as per W3C Atom specification
- **JSON Feed**: Follows JSON Feed 1.1 specification for language support
- **RSS**: Respects RSS 2.0 limitations (channel-level only)

### Testing

Created dedicated test files for maintainability:

- atom1.language.test.ts
- json.language.test.ts
- rss2.language.test.ts

Test Coverage Includes:

- Feed-level and item-level language support
- Multiple language codes (en, fr, es-ES, zh-CN, ja-JP, pt-BR, etc.)
- Security testing with malicious input
- Edge cases and validation
- Format-specific behavior verification

### Language Support Matrix

| Format | Feed-Level | Item-Level | Implementation Status |
|--------|------------|------------|---------------------|
| **Atom 1.0** | ✅ `xml:lang` | ✅ `xml:lang` | **Complete** |
| **JSON Feed 1.1** | ✅ `language` | ✅ `language` | **Complete** |  
| **RSS 2.0** | ✅ `<language>` | ❌ Spec limitation | **Complete** |

#### Basic Feed-Level Language
```typescript
const feed = new Feed({
  title: "My Blog",
  language: "en-US", // Now supported across all formats
  // ... other options
});
```

#### Per-Item Language Support
```typescript
feed.addItem({
  title: "Article en Français",
  language: "fr-FR", // Atom & JSON Feed only
  // ... other item properties
});
```

### Related Issues
- Closes #196 (language should be supported in atom feeds)
- Enhances overall feed internationalization support

### CODE REVIEW

File: src/atom1.ts [hunk 1]
```diff
@@ -11,10 +11,17 @@ import { sanitize } from "./utils";
 export default (ins: Feed) => {
   const { options } = ins;
 
+  const feedAttrs: Record<string, string> = { xmlns: "http://www.w3.org/2005/Atom" };
+  if (ins.options.language) {
+    // Atom uses the reserved "xml:" namespace for language;
+    // no extra xmlns declaration is required.
+    feedAttrs["xml:lang"] = sanitize(ins.options.language) || ins.options.language;
```
Comment (Greenheart) [inline src/atom1.ts]: Why fall back to the unsanitized value if the sanitization fails? Wouldn't it be better to throw an error, or log a warning and omit the `language` field instead?

**Review by Greenheart — COMMENTED** [review]
Thanks for implementing this! I was just about to do this myself, but was happily surprised it was already done :smile:

Tests look good, pass, and the code looks good.

Some comments about related tasks, but some of them might be better suited for separate issues and PRs.
